pandas/core/missing.py

```diff
@@ -248,6 +259,13 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
         except ImportError:
             raise ImportError("Your version of Scipy does not support "
                               "Akima interpolation.")
+    elif method == 'piecewise_polynomial':  # GH12887
```

Comment:
so move this to a separate function which you then use here (and above), something like:

```
def _piecewise_polynomial_compat():
    import scipy
    if LooseVersion(scipy.__version__) > 0.17:
         return scipy.interpolate.from_derivatves
    return scipy.interpolate.piecewise_polonomial_interpolate
```